### DISCUSSION TITLE
error while running npm i -g  truffle@">=5.0.0 <6.0.0"

### DISCUSSION THREAD

**Discussion by maxeq** [ConsenSys-archive/truffle#5845] [General Q&A]
[Truffle: Execute command] 
Working dir: d:\Games\PycharmProjects\eth
Running command
npm i -g  truffle@">=5.0.0 <6.0.0"
[Truffle: Execute command] npm
[Truffle: Execute command]  WARN deprecated mkdirp-promise@5.0.1: This package is broken and no longer maintained. 'mkdirp' itself supports promises now, please switch to that.
[Truffle: Execute command] npm 
[Truffle: Execute command] WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
[Truffle: Execute command] npm WARN 
[Truffle: Execute command] deprecated har-validator@5.1.5: this library is no longer supported
[Truffle: Execute command] npm WARN 
[Truffle: Execute command] deprecated multibase@0.6.1: This module has been superseded by the multiformats module
[Truffle: Execute command] npm WARN deprecated multicodec@0.5.7: This module has been superseded by the multiformats module
[Truffle: Execute command] npm WARN 
[Truffle: Execute command] deprecated apollo-server-errors@3.3.1: The `apollo-server-errors` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm 
[Truffle: Execute command] WARN deprecated apollo-datasource@3.3.2: The `apollo-datasource` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN
[Truffle: Execute command]  deprecated apollo-server-types@3.7.1: The `apollo-server-types` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN 
[Truffle: Execute command] deprecated apollo-server-plugin-base@3.7.1: The `apollo-server-plugin-base` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN 
[Truffle: Execute command] deprecated apollo-server-express@3.11.1: The `apollo-server-express` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN 
[Truffle: Execute command] deprecated apollo-server@3.11.1: The `apollo-server` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN
[Truffle: Execute command]  deprecated apollo-reporting-protobuf@3.3.3: The `apollo-reporting-protobuf` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/usage-reporting-protobuf` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN deprecated
[Truffle: Execute command]  apollo-server-env@4.2.1: The `apollo-server-env` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/utils.fetcher` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Truffle: Execute command] npm WARN deprecated multibase@0.7.0: This module has been superseded by the multiformats module
[Truffle: Execute command] npm 
[Truffle: Execute command] WARN deprecated uuid@2.0.1: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[Truffle: Execute command] npm WARN
[Truffle: Execute command]  deprecated multicodec@1.0.4: This module has been superseded by the multiformats module
[Truffle: Execute command] npm WARN deprecated
[Truffle: Execute command]  cids@0.7.5: This module has been superseded by the multiformats module
[Truffle: Execute command] npm 
[Truffle: Execute command] WARN deprecated apollo-server-core@3.11.1: The `apollo-server-core` package is part of Apollo Server v2 and v3, which are now deprecated (end-of-life October 22nd 2023). This package's functionality is now found in the `@apollo/server` package. See https://www.apollographql.com/docs/apollo-server/previous-versions/ for more details.
[Truffle: Execute command] 
changed 759 packages, and audited 788 packages in 2m
[Truffle: Execute command] 
90 packages are looking for funding
  run `npm fund` for details
[Truffle: Execute command] 
found 0 vulnerabilities
[Truffle: Execute command] Finished running command
[Truffle: Execute command] Traceback (most recent call last):
  File "D:\Programming\Python310\lib\runpy.py", line 196, in _run_module_as_main
[Truffle: Execute command]     return _run_code(code, main_globals, None,
  File "D:\Programming\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
[Truffle: Execute command]   File "D:\Programming\Python310\Scripts\truffle.exe\__main__.py", line 4, in 
[Truffle: Execute command]   File "D:\Programming\Python310\lib\site-packages\truffle\app.py", line 33
    print file_index
    ^^^^^^^^^^^^^^^^
[Truffle: Execute command] SyntaxError: Missing parentheses in call to 'print'. Did you mean print(...)?

**Comment by cipaterson** [comment]
Hi @maxeq,
can you give more context?  What comes before the log output you have provided?

Have you followed https://github.com/orgs/trufflesuite/discussions/5089 ?
Chris

**Comment by eggplantzzz** [comment]
I actually don't really see any errors in the output you have provided. There are some warnings but these should not prevent it from installing successfully.
